File: sort-colors/sort-colors.go

func sortColors(nums []int)  {
    i , j , k := 0,0,len(nums) - 1
    for ;j <= k; {
        if nums[j] == 0 {
            nums[j],nums[i] = nums[i],nums[j]
            j+=1
            i+=1
        }else if nums[j] == 1 {
            j+=1;
        }else{
            nums[j],nums[k] = nums[k],nums[j]
            k-=1
        }
    }
}

File: 75-sort-colors/sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int start = 0, cur = 0, end = nums.size() - 1;
        while(cur <= end){
            if(nums[cur] == 0){
                swap(nums[start], nums[cur]);
                if(cur == start)cur++;
                start++;
            }else if(nums[cur] == 1){
                cur++;
            }else if(nums[cur] == 2){
                 swap(nums[end], nums[cur]);
                 end--;
            }
        }
    }
};